azext_iot/central/services/device.py

```diff
@@ -84,6 +85,47 @@ def list_devices(
     return devices
 
 
+def get_device_registration_summary(
+    cmd, app_id: str, token: str, central_dns_suffix="azureiotcentral.com",
+):
+    """
+    Get device registration summary for a given app
+
+    Args:
+        cmd: command passed into az
+        app_id: name of app (used for forming request URL)
+        token: (OPTIONAL) authorization token to fetch device details from IoTC.
+            MUST INCLUDE type (e.g. 'SharedAccessToken ...', 'Bearer ...')
+        central_dns_suffix: {centralDnsSuffixInPath} as found in docs
+
+    Returns:
+        registration summary
+    """
+
+    registration_summary = {
```

Comment:
for safety, I'd recommend moving this over to a dictionary comprehension. Reason being - if we add a new enum value tomorrow we won't have to update this code due to pythonic magic.

`registration_summary = {status.value: 0 for status in DeviceStatus}`

Reply:
So the reason this works is because an Enum in python is iterable. Its pretty nifty

---
In reply to: [428986126](https://github.com/Azure/azure-iot-cli-extension/pull/192#discussion_r428986126) [](ancestors = 428986126)